src/Utils/dispatchUtils.ts

```diff
@@ -238,28 +269,90 @@ function randomSingleTransfer(
     const dialogTransitionGroups = generateRandomDialogTransitionGroups(modelsTrainDialogs, getPercentageOfRoundsToTransferAt)
     const dialogsWithModelTransition = concatTransitionDialogsWithOtherDialogs(dialogTransitionGroups)
-        .map(generateDispatchDialog)
+        .map(t => generateDispatchDialog(t.rounds))
 
     return [
         ...dialogsWithoutModelTransition,
         ...dialogsWithModelTransition
     ]
 }
 
-const generateDispatchDialog = (t: CLM.TrainDialog) => ({
-    tags: [`generated`],
-    description: "",
-    trainDialogId: uuid(),
-    rounds: t.rounds,
-
-    // Ignored fields (Irrelevant for Dispatch function)
-    clientData: {
-        importHashes: []
-    },
-    initialFilledEntities: [],
-    createdDateTime: new Date().toJSON(),
-    lastModifiedDateTime: new Date().toJSON(),
-} as unknown as CLM.TrainDialog)
+/**
+ * Random Multi-Transfer Dialogs
+ *
+ * @param modelsTrainDialogs Dialogs per models
+ * @param numberOfTransitionsPerDialog Number of Transitions per Dialog
+ * @param numberOfDialogs Number of Dialogs
+ */
+function randomMultiTransfer(
+    modelsTrainDialogs: CLM.TrainDialog[][],
+    numberOfTransitionsPerDialog: number,
+    numberOfDialogs: number,
+): CLM.TrainDialog[] {
+    const trainDialogs: CLM.TrainDialog[] = []
+    // Server limitation of 100 rounds
+    const maxRounds = 100
+    // Could expose roundsPerDialog and calculate roundsPerTransition from the desired length but thought that was harder to imagine
+    const numberOfRoundsPerTransition = Math.floor(maxRounds / numberOfTransitionsPerDialog)
+    const numberOfRoundsPerDialog = maxRounds - numberOfRoundsPerTransition
+
+    while (trainDialogs.length < numberOfDialogs) {
+        let rounds: CLM.TrainRound[] = []
+        let previousModelIndex: number = 0
+
+        while (rounds.length < numberOfRoundsPerDialog) {
+            // Pick new model to transition to, must not be same as previous
+            let modelIndex: number
+            do {
+                modelIndex = Math.floor(Math.random() * modelsTrainDialogs.length)
+            }
+            while (modelIndex === previousModelIndex)
+
+            // Get random dialog from model
+            const dialogsInModel = modelsTrainDialogs[modelIndex]
+            const dialogIndex = Math.floor(Math.random() * dialogsInModel.length)
+            const dialog = dialogsInModel[dialogIndex]
+
+            // Get random N consecutive rounds from dialog
+            // Substract from N to ensure there is space for slice, otherwise use at least 1 round
+            const maxIndex = Math.max(1, dialog.rounds.length - numberOfRoundsPerTransition)
+            const startRoundIndex = Math.floor(Math.random() * maxIndex)
+            let newRounds = dialog.rounds.slice(startRoundIndex, numberOfRoundsPerTransition)
+            // TODO: Decide if there is better alternative. Dialog could not have as many rounds as requested but still be valid
```

Comment:
The part I was mentioning about repeating when there are less than desired number of rounds.
In other words dialog has 9 rounds. Algorithm randomly selects position 6 to start and take 5 more rounds. There are only 3 so it just repeats [6,7,8,6,7].